test/engine/SpatialJoinTest.cpp

```diff
@@ -0,0 +1,1957 @@
+#include <gtest/gtest.h>
+
+#include <cstdlib>
+
+#include "../util/IndexTestHelpers.h"
+#include "./../../src/global/ValueId.h"
+#include "./../../src/util/GeoSparqlHelpers.h"
+#include "engine/ExportQueryExecutionTrees.h"
+#include "engine/IndexScan.h"
+#include "engine/Join.h"
+#include "engine/QueryExecutionTree.h"
+#include "engine/SpatialJoin.h"
+#include "parser/data/Variable.h"
+
+using namespace ad_utility::testing;
```

Comment:
All the global helpers that only appear in a `.cpp`file should be in the anonymous namespace

```
namespace {
// functions and variables here
}
```

This makes them inaccessible outside this .cpp files and avoids linker problems.

Reply:
i think i have done this (but named the namespace localTestHelpers). I'm not completely sure, if i have done this correctly. I have (already before this comment) created a lot of namespaces to structure the test file. The only functions which weren't contained in a namespace, where the ones in the beginning. The reason for the many namespaces is the following: if you are developing a test for a certain function, you can collapse all namespaces, which aren't required for this function, which minimizes the amount of scrolling through the source code

Reply:
We just found out, that you can do both: (and should)
```
namespace { // avoid linking problems
namespace localTestHelpers { // because the file is long
// put stuff here
void f() {return 42;}
}
}

// outside of namspace, you can use localTestHel0pers:::f()

Reply:
Maybe the `TEST(...)` macros need to be outside the anonymouse namespace to work correctly.